### PR TITLE
CC Libraries - add new CLSO layer through placeEvent handle

### PR DESCRIPTION
**Pull request by shaoshing**
The PR change the behavior on handling new CLSO layer and expanded graphic layers (drop graphic asset with ALT modifier).

For CLSO, the new layer is added through the "placeEvent" handler, to keep consistent with embedded layer (show the new layer after transform mode). It also help address issues #2534,  #2524, and #2523 (press delete / esc during transform mode)

lib-so

For expanded graphic layers, since there is no "placeEvent" emitted, we will use the previous way by calling "getLayerIDs" and "addLayers" to discover the new layers.

### CODE REVIEW

File: src/js/actions/libraries.js [hunk 1]
```diff
@@ -793,28 +794,35 @@ define(function (require, exports) {
...
+                        // Dropping graphic asset with ALT/OPT modifier will expand the asset (instead of adding 
+                        // CLSO layer) and result in creating multiple layers in Photoshop. Unlike regualr drop 
+                        // event (without modifier), this will not trigger the "placeEvent" handler, so we need 
+                        // to get the new layer IDs through the difference between the previous layer IDs and the
+                        // next layer IDs.
+                        //
+                        // FIXME: we should instead get IDs back from Photoshop when layers are placed with modifier, 
+                        //        so we don't have to get all the layer IDs.
+                        if (hasAlt) {
+                            placeElementPromise.then(function () {
```
Comment (volfied) [inline src/js/actions/libraries.js]: Not a fan of the structure here. If alt is being pressed, you wait for that promise to fdinish, and then update the document, but you return the placeElement promise already.

What you can do instead is: 

`placeElementPromise = placeElementPromise.then(function ...`

and `return placeElementPromise` as usual at the end.

**Comment by volfied** [comment]
Got one nit @shaoshing, looks good otherwise.

**Comment by shaoshing** [comment]
Thanks @volfied that's a good one! I've made the change so back to you.

**Comment by volfied** [comment]
:+1:

**Pull request merged**